src/Agent.Plugins/TestFilePublisher/Plugin/TestFilePublisherLogPlugin.cs

```diff
@@ -0,0 +1,208 @@
+                _telemetry.AddOrUpdate("PluginDisabledReason", "NoSteps");
+                return true;
+            }
+
+            if (context.Steps.Any(x => x.Id.Equals(new Guid("0B0F01ED-7DDE-43FF-9CBB-E48954DAF9B1"))))
+            {
+                _telemetry.AddOrUpdate("PluginDisabledReason", "ExplicitPublishTaskPresent");
+                return true;
+            }
+
+            if (_pipelineConfig.BuildId == 0)
+            {
+                _telemetry.AddOrUpdate("PluginDisabledReason", "BuildIdZero");
+                return true;
+            }
+
+            if (_pipelineConfig.Pattern == null)
+            {
+                _telemetry.AddOrUpdate("PluginDisabledReason", "PatternIsEmpty");
+                return true;
+            }
+
+            if (_pipelineConfig.SearchFolders == null)
+            {
+                _telemetry.AddOrUpdate("PluginDisabledReason", "SearchFolderIsEmpty");
+                return true;
+            }
+
+            return false;
+        }
+
+        private async Task PopulatePipelineConfig(IAgentLogPluginContext context)
+        {
+            var props = new Dictionary<string, Object>();
+
+            if (context.Variables.TryGetValue("system.teamProject", out var projectName))
+            {
+                _pipelineConfig.ProjectName = projectName.Value;
+                _telemetry.AddOrUpdate("ProjectName", _pipelineConfig.ProjectName);
+                props.Add("ProjectName", _pipelineConfig.ProjectName);
+            }
+
+            if (context.Variables.TryGetValue("build.buildId", out var buildId))
+            {
+                _pipelineConfig.BuildId = int.Parse(buildId.Value);
+                _telemetry.AddOrUpdate("BuildId", _pipelineConfig.BuildId);
+                props.Add("BuildId", _pipelineConfig.BuildId);
+            }
+
+            if (context.Variables.TryGetValue("System.DefinitionId", out var buildDefinitionId))
+            {
+                _telemetry.AddOrUpdate("BuildDefinitionId", buildDefinitionId.Value);
+            }
+
+            if (context.Variables.TryGetValue("agent.testfilepublisher.pattern", out var pattern)
+                && !string.IsNullOrWhiteSpace(pattern.Value))
+            {
+                _pipelineConfig.Pattern = pattern.Value;
+            }
+
+            if (context.Variables.TryGetValue("agent.testfilepublisher.searchfolders", out var searchFolders)
+                && !string.IsNullOrWhiteSpace(searchFolders.Value))
+            {
+                PopulateSearchFolders(context, searchFolders.Value);
+            }
+
+            // Publish the initial telemetry event in case we are not able to fire the cumulative one for whatever reason
+            await _telemetry.PublishTelemetryAsync("TestFilePublisherInitialize", props);
+        }
+
+        private void PopulateSearchFolders(IAgentLogPluginContext context, string searchFolders)
+        {
+            var folderVariables = searchFolders.Split(",");
```

Comment:
can we do same thing for pattern here instead of splitting by comma later?

Reply:
+1/

---
In reply to: [263670515](https://github.com/Microsoft/azure-pipelines-agent/pull/2141#discussion_r263670515) [](ancestors = 263670515)